docs/codeql/codeql-cli/about-codeql-packs.rst

```diff
@@ -56,49 +59,286 @@ The following properties are supported in ``qlpack.yml`` files.
+
+The ``suites`` property indicates a directory where "well-known" query suites can be found. These suites can be used on the command line by referring to their name only, rather than their full path. For more information about query suites, see ":doc:`Creating CodeQL query suites <creating-codeql-query-suites>`."
+
+CodeQL packs for custom tests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For custom CodeQL packs containing test files, you also need to include an
+``extractor`` property so that the ``test run`` command knows how to create test
+databases. You may also wish to specify the ``tests`` property.
+
+.. include:: ../reusables/test-qlpack.rst
+
+For more information about running tests, see ":doc:`Testing custom queries
+<testing-custom-queries>`."
+
+.. _standard-codeql-packs:
+
+Examples of CodeQL packs in the CodeQL repository
+-------------------------------------------------
+
+Each of the languages in the CodeQL repository has four main CodeQL packs:
+
+- Core library pack for the language, with the :ref:`database schema <codeql-database-schema>`
+  used by the language, and CodeQL libraries, and queries at ``<language>/ql/lib``
+- Core query pack for the language that includes the default queries for the language, along
+  with their query suites at ``<language>/ql/src``
+- Tests for the core language libraries and queries at ``<language>/ql/test``
+- Example queries for the language at ``<language>/ql/examples``
+
+Core library pack
+~~~~~~~~~~~~~~~~~
+
+Here is an example ``qlpack.yml`` file for the `C/C++ analysis libraries
+<https://github.com/github/codeql/blob/main/cpp/ql/lib/qlpack.yml>`__
+core language pack:
+
+.. code-block:: yaml
+
+   name: codeql/cpp-all
+   version: x.y.z-dev
+   dbscheme: semmlecode.cpp.dbscheme
+   library: true
+   upgrades: upgrades
+
+Some extra notes on the following properties:
+
+- ``library``: Indicates that this is a library pack with no executable queries. It is only meant to be used as a dependency for other packs.
+- ``dbscheme`` and ``upgrades``: These properties are internal to the CodeQL CLI and should only be defined in the core QL pack for a language.
+
+.. _standard-codeql-query-packs:
+
+Core query pack
+~~~~~~~~~~~~~~~
+
+Here is an example ``qlpack.yml`` file for `C/C++ analysis queries
+<https://github.com/github/codeql/blob/main/cpp/ql/src/qlpack.yml>`__
+core query pack:
+
+.. code-block:: yaml
+
+   name: codeql/cpp-queries
+   version: x.y.z-dev
+   dependencies:
+       codeql/cpp-all: "*"
+       codeql/suite-helpers: "*"
+   suites: codeql-suites
+   defaultSuiteFile: codeql-suites/cpp-code-scanning.qls
+
+Some extra notes on the following properties:
+
+- ``dependencies``: This query pack depends on ``codeql/cpp-all`` and ``codeql/suite-helpers``. Since these dependencies are resolved from source, it does not matter what version of the CodeQL pack they are compatible with.
```

Comment:
I'm assuming that "resolved from source" becomes more obvious when we merge the PR about the CodeQL workspace.

Reply:
Since that article does not exist in this PR, I will add the link in the followup PR where the article is introduced.